### ISSUE TITLE
🐛 [sambanas] 0x80070032 The request is not supported

### ISSUE THREAD

**Issue by Nico04** [dianlight/hassio-addons#182]
### Addon

SambaNAS

### Description

Error `0x80070032 The request is not supported` is displayed by Windows when trying to copy a file.

The only difference between working file and not working one I found is the file attribute. The one that can be copied is "Archive", and the one that can't is 524320.
After further investigation, I found that 524320 is bitmask for "Archive" AND "Pinned" (enum value is 52428, https://learn.microsoft.com/en-us/windows/win32/fileio/file-attribute-constants).
So using powershell command I set the file attribute to "Archive" only, and then the file can be copied normally, without error.

Using Windows 11.

### Reproduction steps

```bash
1. Download and extrat files from MediaInfo software :
https://mediaarea.net/download/binary/mediainfo-gui/23.10/MediaInfo_GUI_23.10_Windows_x64_WithoutInstaller.7z
2. Copy folder to share
```

### Addon Logs

```bash
smb_set_file_dosmode: file_set_dosmode of _Test TO DELETE/MediaInfo_GUI_23.10_Windows_x64_WithoutInstaller/curl-ca-bundle.crt failed: Function not implemented
```

### Addon Config

```bash
workgroup: WORKGROUP
username: homeassistant
allow_hosts:
  - 10.0.0.0/8
  - 172.16.0.0/12
  - 192.168.0.0/16
  - 169.254.0.0/16
  - fe80::/10
  - fc00::/7
automount: true
moredisks: []
mountoptions:
  - nosuid
  - relatime
  - noexec
available_disks_log: false
medialibrary:
  enable: false
veto_files:
  - ._*
  - .DS_Store
  - Thumbs.db
  - icon?
  - .Trashes
compatibility_mode: false
recyle_bin_enabled: false
wsdd2: false
mqtt_use_legacy_entities: false
autodiscovery: {}
other_users: []
acl:
  - share: addons
    disabled: true
  - share: config
    disabled: true
  - share: ssl
    disabled: true
interfaces: []
password: ***
```

### Architecture

other

### OS

HaOS

**Comment by dianlight** [comment]
What is your drive and source file system type?

If destination drive fs don't supports extended attributes ( is exfat or similar) disable the time machine option on the drive and it should works.

L

**Comment by dianlight** [comment]
Fixed in 12.2.0-nas #191
